src/test/lib/concurrency/stress_test.cpp

```diff
@@ -263,6 +276,178 @@ TEST_F(StressTest, NodeSchedulerStressTest) {
+      waiting_jobs.emplace_back(std::make_shared<JobTask>([&] {
+        while (wait_flag) {
+          std::this_thread::sleep_for(SLEEP_TIME);
+        }
+        ++counter;
+      }));
+      // We create runs of dependent jobs and set the current job as a predecessor of the previous job.
+      if (job_id % DEPENDENT_JOB_TASKS_LENGTH != 0) {
+        waiting_jobs.back()->set_as_predecessor_of(waiting_jobs[job_id - 1]);
+      }
+    }
+
+    for (const auto& queue : node_queue_scheduler->queues()) {
+      if (!queue) {
+        continue;
+      }
+      EXPECT_EQ(queue->semaphore.availableApprox(), 0);
+    }
+
+    Hyrise::get().scheduler()->schedule_tasks(waiting_jobs);
+    // Wait a bit for workers to pull jobs and decrement semaphore.
+    std::this_thread::sleep_for(5 * CORES_PER_NODE * SLEEP_TIME);
+
+    for (const auto& queue : node_queue_scheduler->queues()) {
+      if (!queue) {
+        continue;
+      }
+      EXPECT_EQ(queue->semaphore.availableApprox(),
+                static_cast<size_t>(
+                    std::ceil(static_cast<float>(job_count) / static_cast<float>(DEPENDENT_JOB_TASKS_LENGTH))) -
+                    CORES_PER_NODE);
+    }
+
+    wait_flag = false;
+    Hyrise::get().scheduler()->wait_for_tasks(waiting_jobs);
+
+    for (const auto& queue : node_queue_scheduler->queues()) {
+      if (!queue) {
+        continue;
+      }
+      EXPECT_EQ(queue->semaphore.availableApprox(), 0);
+    }
+    EXPECT_EQ(job_count, counter);
+  }
+}
+
+// The issue with active NUMA nodes (see #2548) did not occur in our non-NUMA-bound CI setup but only when executing
+// NUMA-bound benchmarks on a server. To catch such issues, this test executes a comparatively complex TPC-H query for
+// different fake NUMA topologies.
+TEST_F(StressTest, NodeQueueSchedulerMultiNumaNodeTPCHQ13) {
+  const auto benchmark_config = BenchmarkConfig::get_default_config();
+
+  TPCHTableGenerator(0.1f, ClusteringConfiguration::None, std::make_shared<BenchmarkConfig>(benchmark_config))
+      .generate_and_store();
+
+  auto topologies = FAKE_SINGLE_NODE_NUMA_TOPOLOGIES;
+  topologies.insert(topologies.end(), FAKE_SINGLE_NODE_NUMA_TOPOLOGIES.begin(), FAKE_SINGLE_NODE_NUMA_TOPOLOGIES.end());
+
+  const auto tpch_q13 = std::string{
+      "SELECT c_count, count(*) as custdist FROM (SELECT c_custkey, count(o_orderkey) "
+      "as c_count FROM customer left outer join orders on c_custkey = o_custkey AND "
+      "o_comment not like '%special%request%' GROUP BY c_custkey) as c_orders "
+      "GROUP BY c_count ORDER BY custdist DESC, c_count DESC;"};
+
+  for (const auto& fake_numa_topology : FAKE_SINGLE_NODE_NUMA_TOPOLOGIES) {
+    Hyrise::get().topology.use_fake_numa_topology(fake_numa_topology);
+    auto node_queue_scheduler = std::make_shared<NodeQueueScheduler>();
+    Hyrise::get().set_scheduler(node_queue_scheduler);
+
+    auto sql_pipeline = SQLPipelineBuilder{tpch_q13}.create_pipeline();
+    sql_pipeline.get_result_tables();
```

Comment:
Do we want to assert sth here, e.g., that the status was successful? Not that it would be required in this test case.